calculate-regions.wdl

```diff
@@ -0,0 +1,91 @@
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import "tasks/bedtools.wdl" as bedtools
+
+workflow CalculateRegions {
+    input {
+        File referenceFastaFai
+        File XNonParRegions
+        File YNonParRegions
+        File? regions
+        Map[String, String] dockerImages = {
+              "bedtools": "quay.io/biocontainers/bedtools:2.23.0--hdbcaa40_3"
+        }
+    }
+    call bedtools.MergeBedFiles as mergeBeds {
+        input:
+            bedFiles = [XNonParRegions, YNonParRegions],
+            dockerImage = dockerImages["bedtools"]
+    }
+    # We define the 'normal' regions by creating a regions file that covers
+    # everything except the XNonParRegions and the YNonParRegions.
+    call bedtools.Complement as inverseBed {
+        input:
+            inputBed = mergeBeds.mergedBed,
+            faidx = referenceFastaFai,
+            outputBed = "autosomal_regions.bed",
+            dockerImage = dockerImages["bedtools"]
+    }
+
+    if (defined(regions)) {
+        call bedtools.Intersect as intersectAutosomalRegions {
+            input:
+                regionsA = inverseBed.complementBed,
+                regionsB = select_first([regions]),
+                faidx = referenceFastaFai,
+                outputBed = "intersected_autosomal_regions.bed",
+                dockerImage = dockerImages["bedtools"]
+        }
+
+        call bedtools.Intersect as intersectX {
+            input:
+                regionsA = XNonParRegions,
+                regionsB = select_first([regions]),
+                faidx = referenceFastaFai,
+                outputBed = "intersected_x_non_par_regions.bed",
+                dockerImage = dockerImages["bedtools"]
+        }
+
+        call bedtools.Intersect as intersectY {
+            input:
+                regionsA = YNonParRegions,
+                regionsB = select_first([regions]),
+                faidx = referenceFastaFai,
+                outputBed = "intersected_y_non_par_regions.bed",
+                dockerImage = dockerImages["bedtools"]
+        }
+    }
+
+    output {
+        File Xregions = select_first([intersectX.intersectedBed, XNonParRegions])
+        File Yregions = select_first([intersectY.intersectedBed, YNonParRegions])
+        File autosomalRegions = select_first([intersectAutosomalRegions.intersectedBed, inverseBed.complementBed])
+    }
+
+    parameter_meta {
+        regions: {description: "A bed file describing the regions to operate on. Will be used to intersect the other regions.", category: "common"}
+        XNonParRegions: {description: "Bed file with the non-PAR regions of X", category: "common"}
+        YNonParRegions: {description: "Bed file with the non-PAR regions of Y", category: "common"}
```

Comment:
These are required inputs

Reply:
:+1: Thanks. copy-paste error. Will fix this.